devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,206 @@
+			zigbeeMap.name = "fanSpeed"
+			childDevice.sendEvent(zigbeeMap)
+			if (zigbeeMap.value == "00") {
+				log.debug "fan_off => switch off"
+				childDevice.sendEvent(name: "switch", value: "off")
+				childDevice.sendEvent(name: "level", value: 0)	// For cloud to cloud device UI update
+			} else {
+				log.debug "fan_on => switch on"
+				childDevice.sendEvent(name: "switch", value: "on")
+				def int_v  = zigbeeMap.value as Integer
+				int_v = int_v * 25
+				int_v = int_v > 100 ? 100 : int_v
+				log.debug "child device Level set $int_v"
+				childDevice.sendEvent(name: "level", value: int_v)	// For cloud to cloud device UI update
+			}
+		}		
+	} else {
+		def cluster = zigbee.parse(description)
+		if (cluster && 
+		    cluster.clusterId == ON_OFF_CLUSTER_VALUE &&
+		    cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00) {
+				log.debug "ON/OFF REPORTING CONFIG RESPONSE: " + cluster
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			} else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value) {
+	zigbee.setLevel(value)
+}
+
+def setFanSpeed(speed, device=null) {
+	if (device == null) {
+		return
+	}
+    	log.debug "parent setFanSpeed"
+    	if (speed as Integer == 0) {		
+        	log.debug "fan_off"
+	    	sendFanSpeed(0x00)
+	} else if (speed as Integer == 1) {		
+        	log.debug "low"
+	    	sendFanSpeed(0x01)
+	} else if (speed as Integer == 2) {		
+        	log.debug "medium"	
+        	sendFanSpeed(0x02)    
+	} else if (speed as Integer == 3) {		
+        	log.debug "high"	
+	    	sendFanSpeed(0x03)
+    	} else if (speed as Integer == 4) {		
+        	log.debug "max"
+	    	sendFanSpeed(0x04)
+    	}
+}
+
+private sendFanSpeed(val) {
+	delayBetween([zigbee.writeAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE, DataType.ENUM8, val), zigbee.readAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE)], 100)
+}
+
+def ping() {
+	// PING is used by Device-Watch in attempt to reach the Device
+	return zigbee.onOffRefresh()
+}
+
+def refresh() {
+	zigbee.onOffRefresh() +
+	zigbee.levelRefresh() +	
+	zigbee.readAttribute(FAN_CLUSTER_VALUE, FAN_STATUS_VALUE)
+}
+
+def configure() {
+	log.debug "Configuring Reporting and Bindings."
+	// Device-Watch allows 2 check-in misses from device + ping (plus 1 min lag time)
+	// enrolls with default periodic reporting until newer 5 min interval is confirmed
+	sendEvent(name: "checkInterval", value: 2 * 10 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+	// OnOff minReportTime 0 seconds, maxReportTime 5 min. Reporting interval if no activity
+	zigbee.onOffConfig(0, 300)
+	zigbee.levelConfig()
+	refresh()
+}
+
+def installed() {
+	log.debug "Samsung ITM test prod installed"
+	addChildFan()    
+	configure()
```

Comment:
No need to call it, since you've declared `Configuration` capability.

Reply:
Oh, I didn't know that. Thank you.